corehq/blobs/tests/test_mixin.py

```diff
@@ -378,7 +381,74 @@ def open(self):
 
         def listdir(self):
             summaries = self.s3_bucket.objects.filter(Prefix=self.path + "/")
-            return [o.key for o in summaries]
+            try:
+                return [o.key for o in summaries]
+            except ClientError as err:
+                if not is_not_found(err):
+                    raise
+            return []
+
+
+class TestBlobMixinWithMigratingDbBeforeCopyToNew(TestBlobMixinWithS3Backend):
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestBlobMixinWithMigratingDbBeforeCopyToNew, cls).setUpClass()
+        cls.db = PutInOldBlobDB(cls.db, TemporaryFilesystemBlobDB())
+
+    class TestBlob(TestBlobMixinWithS3Backend.TestBlob):
+
+        def __init__(self, db, name, bucket):
+            self.db = db.new_db
+            self.path = db.new_db.get_path(name, bucket)
+            self.fspath = db.old_db.get_path(name, bucket)
+
+        @property
+        def super(self):
```

Comment:
be careful with it—it occurred to me just now that you wouldn't want to do this in a complex class hierarchy because a subclass accessing it will likely not get expected result. I did it because I was annoyed with the length of the boilerplate, but in general I think it's an anti-pattern. Wouldn't have this problem with [Python 3's `super()`](https://docs.python.org/3/library/functions.html#super).